rep-0149.rst

```diff
@@ -505,10 +494,49 @@ Attributes
   
     "You must give any other recipients of the Work or Derivative Works a copy of this License"
 
+ ``source-files="FILENAME-PATTERN"`` *(optional)*
+
+  A filename pattern using the simplified shell glob syntax specified in
+  Section 6.9 of the `Machine-readable debian/copyright file specification V1.0
+  <https://www.debian.org/doc/packaging-manuals/copyright-format/1.0/>`_ and
+  relative to the ``package.xml`` file.
+
+  The filename pattern specifies the source files this license information
+  refers to. The value ``source-files="*"`` refers to all source files of the
+  package, including source files that are downloaded automatically during the
+  build process - for example in the case of so-called *vendor packages*. If
+  the attribute is not specified, the tag again refers to all source files of
+  the package, including downloaded source files.
+
+  If the filename patterns of multiple license tags match a particular file,
+  the last tag applies to it - following the logic described in Section 6.9
+  of the `Machine-readable debian/copyright file specification V1.0
+  <https://www.debian.org/doc/packaging-manuals/copyright-format/1.0/>`_.
+  Consequently, more general tags should be given first.
+
+ ``type="LICENSE-IDENTIFIER-TYPE"`` *(optional)*
```

Comment:
I think it would be better to eliminate this `type` attribute. I haven't seen it in any other package managers. We can still state that the license SHOULD be an SPDX identifier (and linter can throw a warning).

Reply:
I think that this is important to have as a way to suppress linter errors. In particular, I think that we should change it to be spdx is the default type. As that should be our recommendation. But if you have a non-spdx license you can set it to be "freeform" explicitly and then the linter will be suppressed and not complain about the string not matching spdx. 

This will encourage spdx by it being the shortest path to usage. And if they aren't using an spdx license people can be given a linting warning. And the fix for the linter is to either fix their license declaration to be spdx compatible, or declare that it's freeform with the non-default type attribute.

Reply:
If spdx is the default that works 👍

Reply:
I am also in favor of making spdx the standard. @clalancette understandably argued that this would lead to very many error messages for older packages.

Idea: We make freeform the default (as currently stated in this PR and approved by the TSC yesterday) but make a prominent info message in the linter tooling if the type-attribute is not specified explicitly _and_ the license identifier is not from the SPDX list. Furthermore, we raise awareness for the license documentation topic by a Discourse post and a ROSCon talk. Then, in about a year, we can consider changing the default of the type attribute to SPDX (by a new PR to this REP) to force the maintainers of the remaining packages to resolve/clarify the license documentation. What do you think, @amacneil, @clalancette ?

Reply:
+1 for doing a followup/tick tock process to make it the default with strong linting/warning. But lets close this out first.

Reply:
Yes, the way I would do this is the following:

1. Make `freeform` the default for now.
2. Raise awareness with the Discourse post and ROSCon talk.
3. Update the core packages (everything in https://github.com/ros2/ros2/blob/rolling/ros2.repos) to use SPDX.
4. Add in a linter warning in `ament_copyright` for packages that don't specify the type and the identifier is not in SPDX.
5. Later on, consider changing the default to SPDX.